sklearn/feature_selection/tests/test_mutual_info.py

```diff
@@ -168,12 +176,18 @@ def test_mutual_info_classif_mixed():
         assert mi_nn[2] == mi[2]
 
 
-def test_mutual_info_options():
-    X = np.array([[0, 0, 0], [1, 1, 0], [2, 0, 1], [2, 0, 1], [2, 0, 1]], dtype=float)
-    y = np.array([0, 1, 2, 2, 1], dtype=float)
+def test_mutual_info_options(global_dtype):
+    X = np.array(
+        [[0, 0, 0], [1, 1, 0], [2, 0, 1], [2, 0, 1], [2, 0, 1]], dtype=global_dtype
+    )
+    y = np.array([0, 1, 2, 2, 1])
     X_csr = csr_matrix(X)
 
-    for mutual_info in (mutual_info_regression, mutual_info_classif):
+    for mutual_info, y_dtype in (
+        (mutual_info_regression, global_dtype),
+        (mutual_info_classif, y.dtype),
+    ):
+        y = y.astype(global_dtype)
```

Comment:
I think the `assert_array_equal` occurrences of this tests should be replaced by our new `assert_allclose` helper just in case the `XXX` comment of line https://github.com/scikit-learn/scikit-learn/pull/22677/files#diff-f49e8e483cf2a803b1bd0d0bfb20e3a7303c8e0fdcf248cfa1a365acbfcf05f1R150-R152 is addressed in a future version of scikit-learn.